packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -1173,7 +1173,9 @@ class ReactDOMServerRenderer {
     } else if (tag === 'option') {
       let selected = null;
       const selectValue = this.currentSelectValue;
-      const optionChildren = flattenOptionChildren(props.children);
+      const optionChildren = Array.isArray(props.children)
+        ? flattenOptionChildren(props.children)
+        : props.children;
```

Comment:
Checking for an Array isn't sufficient since React [supports using iterables as children](https://github.com/facebook/react/blob/master/packages/react-dom/src/__tests__/ReactDOMServerIntegrationBasic-test.js#L113-L134). Instead, can we invert the logic so that it checks for `undefined`?

```js
const optionChildren = props.children === undefined
    ? props.children
    : flattenOptionChildren(props.children)
```

Alternatively we could just add that check to `flattenOptionChildren` directly

Reply:
You're totally right. I'm not sure all of the possible cases here. I do know that `null` is also possible in this case. Would `null` and `undefined` checks be sufficient?

Reply:
>In that case we still want to throw because that counts as using children

Do we? On the client we only fire the invariant with `== null` check so neither `null` nor `undefined` counts:

https://github.com/facebook/react/blob/571a9208d5133e8737f565fe60b762d201f0d37c/src/renderers/dom/shared/ReactDOMComponent.js#L161-L164

Server logic should be the same, shouldn't it? Or am I missing something?

We should have a test for this too I guess.